addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQDualBedHandler.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sleepiq.handler;
+
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sleepiq.config.SleepIQBedConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.syphr.sleepiq.api.SleepIQ;
+import org.syphr.sleepiq.api.model.BedSideStatus;
+import org.syphr.sleepiq.api.model.BedStatus;
+
+/**
+ * The {@link SleepIQDualBedHandler} is responsible for handling channel state updates from the cloud service.
+ *
+ * @author Gregory Moyer - Initial contribution
+ */
+public class SleepIQDualBedHandler extends BaseThingHandler implements BedStatusListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Collections.singleton(THING_TYPE_DUAL_BED);
+
+    private final Logger logger = LoggerFactory.getLogger(SleepIQDualBedHandler.class);
+
+    private volatile String bedId;
+
+    public SleepIQDualBedHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Verifying SleepIQ cloud/bridge configuration");
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No cloud service bridge has been configured");
+            return;
+        }
+
+        if (ThingStatus.ONLINE != bridge.getStatus()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
```

Comment:
I expect this line slightly later on will throw an exception if the bridge did not successfully authenticate (and thus become ONLINE):

```
updateProperties(cloudHandler.updateProperties(cloudHandler.getBed(bedId), editProperties()));
```

because of ```cloudHandler.getBed(bedId)```.

Given that, do you recommend a refactoring or leave it the way it is? I see that the ```HueLightHandler``` is also setting itself offline when the bridge is offline. Is there a mechanism I should be using to recover? It looks like the ```SqueezeBoxPlayerHandler``` is using the method ```bridgeStatusChanged``` - would you recommend that approach to recover when the bridge returns?